cmd/state/main.go

```diff
@@ -251,6 +251,13 @@ func run(args []string, isInteractive bool, cfg *config.Instance, out output.Out
 		}
 	}
 
+	// Centralize handling the lack of commit ID in the project to here.
+	// We cannot throw an error during projectfile parsing, because `state reset` needs a valid
+	// primer project struct to work with.
+	if childCmd != nil && childCmd.Name() != "reset" && pj != nil && !pj.Source().HasCommitID() {
+		return locale.NewInputError("err_bad_project_url_reset")
```

Comment:
I think this error should come from projectfile.Parse(). We can then during prime initialization disregard this error if we detect we are running `reset`. 

Otherwise we are only surfacing errors for the projectfile detected in the primer, which isn't guaranteed to be the projectfile that ends up being used.

Reply:
Sorry, I don't like it. `projectfile.Parse()` always returns `nil, err` on error. In my limited experience with Go, whenever you get an err, you can't use the return value because it's contents are undefined (not literally, but due to the error and where it occurred, the fields, etc. may not be populated as one would expect). In order to do what you're suggesting, inside `projectfile.Parse()` we would have to recognize "non-fatal" errors and continue (but save them/pack them for returning at the end of the function). I'm concerned that that would turn into a spaghetti mess.

Reply:
I did some work in the meantime to prove the point, and it is indeed a spaghetti mess. It's going to be a maintenence nightmare to keep track of when a project parse error is okay/non-fatal vs. not okay/fatal and how to continue. As you can see with status checks, I some seemingly unrelated unit test failures popped up.

Based on your feedback about the project being passed to the primer possibly not ultimately being the target project, I think we ought to abandon this. Is lack of commitID a common enough occurrence to warrant such intervention?